CI test of TensorFlow is failing

## Describe the bug
The following CI test fails: https://github.com/huggingface/datasets/runs/8246722693?check_suite_focus=true
```
FAILED tests/test_py_utils.py::TempSeedTest::test_tensorflow - AssertionError:
```

Details:
```
_________________________ TempSeedTest.test_tensorflow _________________________
[gw0] linux -- Python 3.7.13 /opt/hostedtoolcache/Python/3.7.13/x64/bin/python

self = <tests.test_py_utils.TempSeedTest testMethod=test_tensorflow>

    @require_tf
    def test_tensorflow(self):
        import tensorflow as tf
        from tensorflow.keras import layers
    
        def gen_random_output():
            model = layers.Dense(2)
            x = tf.random.uniform((1, 3))
            return model(x).numpy()
    
        with temp_seed(42, set_tensorflow=True):
            out1 = gen_random_output()
        with temp_seed(42, set_tensorflow=True):
            out2 = gen_random_output()
        out3 = gen_random_output()
    
>       np.testing.assert_equal(out1, out2)
E       AssertionError: 
E       Arrays are not equal
E       
E       Mismatched elements: 2 / 2 (100%)
E       Max absolute difference: 0.84619296
E       Max relative difference: 16.083529
E        x: array([[-0.793581,  0.333286]], dtype=float32)
E        y: array([[0.052612, 0.539708]], dtype=float32)

tests/test_py_utils.py:149: AssertionError
```

--- a/setup.py
+++ b/setup.py
@@ -124,7 +124,7 @@ TESTS_REQUIRE = [
     "moto[s3,server]==2.0.4",
     "rarfile>=4.0",
     "s3fs>=2021.11.1",  # aligned with fsspec[http]>=2021.11.1
-    "tensorflow>=2.3,!=2.6.0,!=2.6.1,<2.10",  # temporarily pin <2.10
+    "tensorflow>=2.3,!=2.6.0,!=2.6.1",
     "torch",
     "torchaudio<0.12.0",
     "soundfile",

--- a/tests/test_py_utils.py
+++ b/tests/test_py_utils.py
@@ -135,8 +135,8 @@ class TempSeedTest(TestCase):
         import tensorflow as tf
         from tensorflow.keras import layers
 
+        model = layers.Dense(2)
         def gen_random_output():
-            model = layers.Dense(2)
             x = tf.random.uniform((1, 3))
             return model(x).numpy()
 

--- a/tests/test_py_utils.py
+++ b/tests/test_py_utils.py
@@ -136,6 +136,7 @@ class TempSeedTest(TestCase):
         from tensorflow.keras import layers
 
         model = layers.Dense(2)
+
         def gen_random_output():
             x = tf.random.uniform((1, 3))
             return model(x).numpy()

